include/shark/ObjectiveFunctions/Benchmarks/MultiObjectiveBenchmark.h

```diff
@@ -0,0 +1,189 @@
+#include <shark/ObjectiveFunctions/Benchmarks/RotatedErrorFunction.h>
+#include <shark/ObjectiveFunctions/BoxConstraintHandler.h>
+#include <shark/Rng/GlobalRng.h>
+
+#include <shark/LinAlg/rotations.h>
+#include <tuple>
+
+namespace shark {
+
+namespace detail{
+	//taken from the web. implements an std::integer_sequence type representing a sequence 0,...,N-1, std::integer_sequence is not here until C++14
+	template<int...> struct integer_sequence { using type = integer_sequence; };
+	template<typename T1, typename T2> struct integer_sequence_concat;
+	template<int... I1, int... I2> struct integer_sequence_concat<integer_sequence<I1...>, integer_sequence<I2...>>: integer_sequence<I1..., (sizeof...(I1) + I2)...> {};
+
+	//generate_integer_sequence generates an integer sequence of integers 0....N. requires log(N) template instantiations
+	template<int N> struct generate_integer_sequence;
+	template<int N> struct generate_integer_sequence: integer_sequence_concat<typename generate_integer_sequence<N/2>::type, typename generate_integer_sequence<N-N/2>::type>::type {};
+	template <> struct generate_integer_sequence<0>: integer_sequence<>{};
+	template <> struct generate_integer_sequence<1>: integer_sequence<0>{};
+};
+
+/// \brief Creates  a multi-objective Benchmark from a set of given single objective functions
+///
+/// A variadic template is used to generate a set of benchmarks.
+/// eg MultiObjectiveBenchmark<Sphere,Ellispoid,Rosenbrock> sets up a three-objective Benchmark.
+///
+/// A random rotation and translation is applied to each benchmark function, thus
+/// MultiObjectiveBenchmark<Sphere,Sphere> forms a non-degenerate front.
+/// the ith objective can be queried via the get<i> member function.
+///
+/// The generated translations are approximately sampled from the unit ball and starting points are also drawn
+/// by the same distribution around a random optimum (assuming the optimum is at (0,0) of the untranslated function
+template<class ... Objectives>
+class MultiObjectiveBenchmark: public MultiObjectiveFunction{
+public:
+	MultiObjectiveBenchmark(std::size_t numVariables = 5){
+		m_features |= CAN_PROPOSE_STARTING_POINT;
+		m_features |= HAS_FIRST_DERIVATIVE;
+		setupRotations(typename detail::generate_integer_sequence<sizeof...(Objectives)>::type());
+		setNumberOfVariables(numVariables);//prevent that different objectives have different default number of variables.
+		for(auto& f: m_rotations){
+			//if one function does not have a first derivative, no function has
+			if(!f.hasFirstDerivative())
+				m_features.reset(HAS_FIRST_DERIVATIVE);
+		}
+	};
+	
+	///\brief Name of the Benchmark
+	///
+	/// The name has the form Objective1/Objective2/Objective3/.../ObjectiveN
+	/// where ObjectiveK is the name of the k-th objective.
+	std::string name()const{
+		return generateName(typename detail::generate_integer_sequence<sizeof...(Objectives)>::type());
+		
+	}
+	
+	bool hasScalableDimensionality()const{
+		return true;
```

Comment:
this should give an exception in the constructor anyways. but i added a meaningful check to make this more clear. As this is a benchmark function it does not really make sense to have fixed dimensionality. Otherwise i would need to find a way to construct the objects in a way that all end up with the same dimensionality.